test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -0,0 +1,287 @@
+
+        log.info("{} generating 2nd stream-cuts for {}/{}", READER_GROUP_2, SCOPE, STREAM);
+        futureCuts2 = readerGroup2.generateStreamCuts(streamCutExecutor);
+        // Wait for 5 seconds to force reader group state update. This will allow for the silent
+        // checkpoint event generated as part of generateStreamCuts to be picked and processed.
+        Exceptions.handleInterrupted(() -> TimeUnit.SECONDS.sleep(5));
+        read2 = reader2.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read2.getEvent());
+        assertTrue("Stream-cut generation did not complete for reader group 2", Futures.await(futureCuts2, 10000));
+        streamCuts2 = futureCuts2.join();
+
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP_1, streamCuts1);
+        readerGroup1.updateRetentionStreamCut(streamCuts1);
+        log.info("{} updating its retention stream-cut to {}", "RG2", streamCuts2);
+        readerGroup2.updateRetentionStreamCut(streamCuts2);
+
+        // Retention set has two stream cut at 0/210, 0/330
+        // READER_GROUP_1 updated stream cut at 0/270, READER_GROUP_2 updated stream cut at 0/300
+        // Subscriber lower bound is 0/270, but since truncating at SLB leaves less data in the stream than minimum limit
+        // So truncating at stream cut 0/210 which leaves more data in the stream than min limit
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        // Check to make sure truncation happened at min stream cut from retention set
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 210.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(SCOPE, STREAM), 0L).join().values().stream().anyMatch(off -> off >= 210),
+                5000, 3 * 60 * 1000L);
+
+        // fill stream with 5 more events
+        for (int i = 1; i <= 5; i++) {
+            log.info("Writing event to {}/{}", SCOPE, STREAM);
+            writer.writeEvent(SIZE_30_EVENT).join();
+        }
+
+        // Retention set has two stream cut at 0/330, 0/480
+        // READER_GROUP_1 updated stream cut at 0/270, READER_GROUP_2 updated stream cut at 0/300
+        // Subscriber lower bound is 0/270, but since truncating at SLB leaves more data in the stream than maximum limit
+        // So truncating at stream cut i.e. 0/330 which leave less data in the stream the maximum limit
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        // Check to make sure truncation happened at max stream cut
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 330.", true, () -> controller.getSegmentsAtTime(
```

Comment:
Same here

Reply:
updated it to `anyMatch(off -> off == 210)`.